R/get_map_points.R

```diff
@@ -8,47 +8,49 @@
-#' @param min_lng \code{integer}. A minimum longitude that represents the
-#'   southwest corner of a bounding box of the points to return. Requires
-#'   `min_lat`, `max_lat`, and `max_lng`. Ignored if `point_id` is supplied.
-#' @param max_lat \code{integer}. A maximum latitude that represents the
-#'   northeast corner of a bounding box of the points to return. Requires
-#'   `min_lat`, `min_lng`, and `max_lng`. Ignored if `point_id` is supplied.
-#' @param max_lng \code{integer}. A maximum longitude that represents the
-#'   northeast corner of a bounding box of the points to return. Requires
-#'   `min_lat`, `min_lng`, and `max_lat`. Ignored if `point_id` is supplied.
+#' @param min_lat \code{integer}. The minimum latitude of the points to return.
+#'   Requires `min_lng`, `max_lat`, and `max_lng`. Ignored if `point_id` is
+#'   supplied.
+#' @param min_lng \code{integer}. A minimum longitude of the points to return.
+#'   Requires `min_lat`, `max_lat`, and `max_lng`. Ignored if `point_id` is
+#'   supplied.
+#' @param max_lat \code{integer}. A maximum latitude of the points to return.
+#'   Requires `min_lat`, `min_lng`, and `max_lng`. Ignored if `point_id` is
+#'   supplied.
+#' @param max_lng \code{integer}. A maximum longitude of the points to return.
+#'   Requires `min_lat`, `min_lng`, and `max_lat`. Ignored if `point_id` is
+#'   supplied.
 #' @param source_id \code{integer}. The unique identification number(s) of the
 #'   source(s) to filter the points by. Ignored if `point_id` is supplied.
 #'
 #' @return An \code{sf} object containing the following columns:
 #' \itemize{
-#'   \item \code{point_id}: The identification number of the outcrop point
-#'     element.
-#'   \item \code{strike}: The strike value of the outcrop point element.
-#'   \item \code{dip}: The dip value of the outcrop point element.
-#'   \item \code{dip_dir}: The dip direction value of the outcrop point
-#'   element.
-#'   \item \code{point_type}: The type of the outcrop point element (e.g.
-#'   bedding).
-#'   \item \code{certainty}: The outcrop point element location certainty.
-#'   \item \code{comments}: Notes assigned to the outcrop point element.
+#'   \item \code{point_id}: The identification number of the point element.
+#'   \item \code{strike}: The strike value of the point element.
+#'   \item \code{dip}: The dip value of the point element.
+#'   \item \code{dip_dir}: The dip direction of the point element.
+#'   \item \code{point_type}: The type of point element (e.g. "bedding").
+#'   \item \code{certainty}: The certainty of the point element location.
+#'   \item \code{comments}: Notes assigned to the point element.
 #'   \item \code{source_id}: The unique identification number of the source
-#'   for the outcrop point element.
+#'     for the point element.
 #'   \item \code{geometry}: The point spatial data.
 #' }
 #'   If `sf` is `TRUE` (the default), an `sf` object is returned instead, with
```

Comment:
This should have been removed by me already, as Will noted, geojson seems to be the only option with this endpoint (which is why I didn't include sf as an argument). We could also have some internal handling to dump the geometries when `sf = FALSE` and just return a `data.frame`. Any preference?

Reply:
I am in favour of consistency, but also we shouldn't have to go out of our way to provide consistency that isn't supported by the API itself. I have no problem with the only option being geojson here, as long as the documentation/code is all internally consistent, at present I don't think it is.

Reply:
A tough one, I'm of the same mindset as you, but perhaps leaning towards consistency. It doesn't look like it would require much more than one or two lines of code, perhaps something like:
```
cbind.data.frame(st_drop_geometry(points), st_coordinates(points))
```
This way we would keep the coordinates, but have a `data.frame`. Happy to implement if you agree!

Reply:
If you are happy to do this, I think it would be the best option. I think this is the last outstanding point, so happy to merge once this is resolved!

Reply:
I've just pushed some changes @bethany-j-allen, let me know what you think!